text/3463-crates-io-policy-update.md

```diff
@@ -0,0 +1,233 @@
+- Start Date: 2023-07-24
+- RFC PR: [rust-lang/rfcs#3463](https://github.com/rust-lang/rfcs/pull/3463)
+
+# Summary
+[summary]: #summary
+
+The Rust community has outgrown the current crates.io policies. This RFC proposes a new "Terms of Use" policy based on prior work by PyPI, npm and GitHub.
+
+# Motivation
+[motivation]: #motivation
+
+> Why are we doing this? What use cases does it support? What is the expected outcome?
+
+crates.io has a "[Package Policies](https://crates.io/policies)" page, which describes the current, organically grown policies. A lot of support requests or questionable uses of crates.io however need explicit decisions from the crates.io team, since many cases are currently not covered by these policies. Additionally, decisions made by the team may be seen as arbitrary without written guidelines.
+
+The situation around name squatting has lately also reached unsustainable levels, and while [namespaces](https://github.com/rust-lang/rfcs/pull/3243) might help with some parts of the problem we would still need policies for name squatting namespaces then.
+
+The main motivation for this RFC is to give the crates.io team a fixed set of rules to determine if a project is using crates.io in a reasonable way, or if the user should get a warning and the project potentially be removed. It is mostly codifying the existing practices of the team, except for being more strict regarding name squatting.
+
+# Proposal
+[guide-level-explanation]: #guide-level-explanation
+
+The following is a proposed new "Terms of Use" policy for crates.io, replacing <https://crates.io/policies> and <https://crates.io/data-access>.
+
+-------------------------------------------------------------------------------
+
+# Terms of Use
+
+**Short version:** _crates.io is a critical resource for the Rust ecosystem,
+which hosts a variety of packages from a diverse group of users. That resource
+is only effective when our users are able to work together as part of a
+community in good faith. While using crates.io, you must comply with our
+Acceptable Use Policies, which include some restrictions on content and conduct
+on crates.io related to user safety, intellectual property, privacy,
+authenticity, and other limitations. In short, be excellent to each other!_
+
+We do not allow content or activity on crates.io that:
+
+- violates the [Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct)
+  of the Rust project
+- is unlawful or promotes unlawful activities, incurring legal liability in the
+  countries the Rust Foundation officially operates in
+- is libelous, defamatory, or fraudulent
+- amounts to phishing or attempted phishing
```

Comment:
I don't think crates.io is responsible for verifying these behaviors, or checking if it's intentional or a bug. It is simply **not possible**.

Like, do you think a stray `rm -rf /usr /lib/XXX` is "fraudulent" or a bug? Eg, https://github.com/MrMEEE/bumblebee-Old-and-abbandoned/issues/123 Same issue exists about "phising attempt" when uploading some arbitrary files which "accidentally" covers your whole `/etc`.

"*promotes* unlawful activities" can also be (maliciously) accused to bittorrent, IPFS implementations, or youtube-dl which *CAN BE* used for ~sharing knowledge~ pirating files. Even if an author "promotes piracy" in crates' docs, I still don't think they should be taken down simply because the promotion. It's nonetheless just a thought crime.

Reply:
> "promotes unlawful activities" can also be (maliciously) accused to bittorrent, IPFS implementations, or youtube-dl which CAN BE used for sharing knowledge pirating files. Even if an author "promotes piracy" in crates' docs, I still don't think they should be taken down simply because the promotion. It's nonetheless just a thought crime.

(IANAL) In the US, [*Brandenburg v. Ohio*](https://en.wikipedia.org/wiki/Brandenburg_v._Ohio) set the legal standard for this sort of thing. By its rules, crate docs that say "piracy is good" would be protected by the First Amendment, but "here's how to use this crate to commit crimes, it's blazingly fast with a pleasant API" would potentially run afoul of the law.